### ISSUE TITLE
Why not use context modelling proposed on the paper?

### ISSUE THREAD

**Issue by BlueCCircles** [biubug6/Pytorch_Retinaface#81]
@biubug6 Thanks for your outstanding work, it is more understandable than the code with mxnet. But I have some questions.
1. When I retrained the model, then test on val dataset, but I can't get the result you provided, easy val is 0.94049, medium val is 0.9284, hard val is 0.8113. If I use the trained model "Resnet50_Final.pth" to test val dataset, I can get the results you got.
2. The paper use context module and deformable convolution, but your code use SSH, why?
Thanks.

**Comment by Sentient07** [comment]
Hi @BlueCCircles , I also observe the same thing mentioned by you. When I re-trained, the performance was quite poor. Further, when I trained on my own dataset, the accuracy was still much worse. Did you resolve this issue? Were you able to implement the Deformable Conv?

**Comment by BlueCCircles** [comment]
Hi @Sentient07, you can change the input image size from 840 to 640 and use five layers FPN, this will improve the accuracy. For Deformable Conv, add it to the last layer of SSH, it will improve the accuracy,  you can refer to https://github.com/4uiiurz1/pytorch-deform-conv-v2.

**Comment by Sentient07** [comment]
Hello, @BlueCCircles thank you for your response, and thanks again for sharing your findings!. I will certainly try it out with my data and will let you know how it goes. Have a safe weekend!

**Comment by Sentient07** [comment]
Hi, @BlueCCircles I implemented your suggestions and I can confirm better results! Do you mind sharing the implementation? If you're not in a position to open-source your code, I will share :) 
Thank you again for your inputs

**Comment by BlueCCircles** [comment]
Hi， @Sentient07  I'm glad you got a good results. Of cause you can share the implementation. I will close this issue.

**Comment by GYC1996** [comment]
> Hi @Sentient07, you can change the input image size from 840 to 640 and use five layers FPN, this will improve the accuracy. For Deformable Conv, add it to the last layer of SSH, it will improve the accuracy,  you can refer to https://github.com/4uiiurz1/pytorch-deform-conv-v2.

Are you replace the backbone, the backbone of mobile0.25 has just 3 stage, or you downsampling the stage3 twice?

**Comment by pkhigh** [comment]
@Sentient07, @BlueCCircles can you share your implementation of 5 layers FPN and deformable convolution?

**Comment by manideep2510** [comment]
Hi
I have trained Retinaface Resnet50 model as it is given in this repo with the following settings,
1. Use PyTorch `DistributedDataparallel` instead of `Dataparallel`
2. Mixed precision training with Pytorch AMP
2. batch_size = 44
3. Epochs = 100
4. LR = 5e-5
5. NO Deformable Convs

And all other parameters same as that in the default config file.

Testing the model at 99th epoch, got the following results on WIDER Face dataset,

| easy | medium | hard |  average |
|:-|:-:|:-:|:-:|
| 94.88 | 93.94 % | 89.75% | 90.63% |

Seems like the numbers given by the authors hold up, and in my case, I even got slightly better results.

**Comment by faderani** [comment]
@Sentient07 Hey, can you share your implementation with 5 layers FPN and deformable convolution?
